test/performance/benchmarks/rollout-probe/continuous/main.go

```diff
@@ -0,0 +1,131 @@
+	ctx, cancel := context.WithTimeout(ctx, *duration+time.Minute)
+	defer cancel()
+
+	// Use the benchmark key created
+	mc, err := mako.Setup(ctx)
+	if err != nil {
+		log.Fatal("Failed to setup mako: ", err)
+	}
+	q, qclose, ctx := mc.Quickstore, mc.ShutDownFunc, mc.Context
+	// Use a fresh context here so that our RPC to terminate the sidecar
+	// isn't subject to our timeout (or we won't shut it down when we time out)
+	defer qclose(context.Background())
+
+	// Wrap fatalf in a helper or our sidecar will live forever.
+	fatalf := func(f string, args ...interface{}) {
+		qclose(context.Background())
+		log.Fatalf(f, args...)
+	}
+
+	// Validate flags after setting up "fatalf" or our sidecar will run forever.
+	if *target == "" {
+		fatalf("Missing flag: -target")
+	}
+
+	// Based on the "target" flag, load up our target benchmark.
+	// We only run one variation per run to avoid the runs being noisy neighbors,
+	// which in early iterations of the benchmark resulted in latency bleeding
+	// across the different workload types.
+	t, ok := targets[*target]
+	if !ok {
+		fatalf("Unrecognized target: %s", *target)
+	}
+
+	// Make sure the target is ready before sending the large amount of requests.
+	if err := performance.ProbeTargetTillReady(t.target.URL, *duration); err != nil {
+		fatalf("Failed to get target ready for attacking: %v", err)
+	}
+
+	// Set up the threshold analyzers for the selected benchmark.  This will
+	// cause Mako/Quickstore to analyze the results we are storing and flag
+	// things that are outside of expected bounds.
+	q.Input.ThresholdInputs = append(q.Input.ThresholdInputs, t.analyzers...)
+
+	// Send 2000 QPS (2 per ms) for the given duration with a 30s request timeout.
+	rate := vegeta.Rate{Freq: 3, Per: time.Millisecond}
+	targeter := vegeta.NewStaticTargeter(t.target)
+	attacker := vegeta.NewAttacker(vegeta.Timeout(30 * time.Second))
+
+	// Create a new aggregateResult to accumulate the results.
+	ar := metrics.NewAggregateResult(int(duration.Seconds()))
+
+	// Start the attack!
+	results := attacker.Attack(targeter, rate, *duration, "load-test")
+LOOP:
+	for {
+		select {
+		case <-ctx.Done():
+			// If we timeout or the pod gets shutdown via SIGTERM then start to
+			// clean thing up.
+			break LOOP
+
+		case res, ok := <-results:
+			if !ok {
+				// Once we have read all of the request results, break out of
+				// our loop.
+				break LOOP
+			}
+			// Handle the result for this request
+			metrics.HandleResult(q, *res, t.stat, ar)
+		}
+	}
+
+	// Walk over our accumulated per-second error rates and report them as
+	// sample points.  The key is seconds since the Unix epoch, and the value
+	// is the number of errors observed in that second.
+	for ts, count := range ar.ErrorRates {
+		q.AddSamplePoint(mako.XTime(time.Unix(ts, 0)), map[string]float64{
+			t.estat: float64(count),
+		})
+	}
+
+	// Commit data to Mako and handle the result.
+	if err := mc.StoreAndHandleResult(); err != nil {
```

Comment:
In case you missed it, if you want to report the error to Slack autoscaling channel, you need to configure it in `performance-tests.sh`

Reply:
not yet.
I want to build the full one first.